src/source/ObjcGenerator.scala

```diff
@@ -417,6 +420,44 @@ class ObjcGenerator(spec: Spec) extends Generator(spec) {
       }
       w.wl
 
+      w.wl("- (NSDictionary *)toDict")
+      w.braced {
+        w.wl("#define _djinni_hide_null_(_o_) ((_o_)?(_o_):([NSNull null]))")
```

Comment:
@mknejp i see. 

i though that doing this with a macro will be more readable, (instead of generating a bunch of repeated code)

also, if i convert the macro into a function, it will be more efficient as `self.<name>` will be called only once, instead of twice (with the macro)